MAVProxy/modules/mavproxy_console.py

```diff
@@ -349,6 +350,21 @@ def mavlink_packet(self, msg):
                     self.say("%s fail" % s)
             self.last_sys_status_health = msg.onboard_control_sensors_health
 
+            # check for any error bits being set:
+            now = time.time()
+            if now - self.last_sys_status_errors_announce > self.mpstate.settings.sys_status_error_warn_interval:
+                for field_num in range(1, 5):
+                    field = "errors_count%u" % field_num
+                    x = getattr(msg, field, None)
+                    if x is None:
+                        self.say("Failed to get field %s" % field)
+                        self.last_sys_status_errors_announce = now
+                        break
+                    if x != 0:
+                        self.last_sys_status_errors_announce = now
+                        self.say("System failure detected.  Land now.")
```

Comment:
need to make the msg shorter. It takes a long time to announce these things, and meanwhile the pilot doesn't have airspeed etc announcements

Reply:
Shortened to "Critical failure"